### DISCUSSION TITLE
STM32H7 - M4: Flash driver : writing to internal flash sometimes raise STRBERR

### DISCUSSION THREAD

**Discussion by mkaricheri** [zephyrproject-rtos/zephyr#76647] [General]
Hello,

We are using a  STM32H755 SoC and running the application from the Bank 2 on M4. The application runs from lower part (896KiB) and write to a partition which is the last 128KiB of the Bank 2 using the flash write() API. Application write code is based on flash_shell sample application. The write work most of the time. However occasionally it encounters a STRBERR  and write fails. However I read following in the reference manual section 4.7.4

![image](https://github.com/user-attachments/assets/bead083e-fa88-4006-a54b-272a94dd14b5)

So add the following workaround which works great based on our testing. We also readback and verify the flash content. So wondering what causes STRBERR. If we go by the reference manual, we shouldn't return error on this error.
My workaround below is currently done only for Bank 2 write, but if a patch is acceptable, this should be done for Bank 1 as well. So would like to know from experts why this happens and if driver can ignore this as in the commit https://github.com/zephyrproject-rtos/zephyr/compare/main...mkaricheri:zephyr:workaround-flash-write-error-strberr

Thanks
Murali Karicheri
S&C Electric company,

**Comment by mkaricheri** [comment]
cc: @erwango @FRASTM @George-Stefan @gautierg-st @ajarmouni-st

**Comment by erwango** [comment]
Not sure why this happens (even if I'd say it could be linked to use of M4 writing to flash as this is the first time this error is reported), but the fix seems legitimate, so don't hesitate to go ahead to push a PR.

**Reply by mkaricheri** [reply]
@erwango I have a PR ready to push. Will do now

**Reply by mkaricheri** [reply]
Pushed a PR https://github.com/zephyrproject-rtos/zephyr/pull/76740
